tests/models/test_dag.py

```diff
@@ -1898,8 +1951,65 @@ def test_dags_needing_dagruns_only_unpaused(self):
         dag_models = DagModel.dags_needing_dagruns(session).all()
         assert dag_models == []
 
-        session.rollback()
-        session.close()
+    def test_dags_needing_dagruns_doesnot_send_dagmodel_with_null_max_active_runs(self):
+        """
+        We check that max_active_runs must not be null for dags
+        being set to scheduler to create dagruns
+        """
+        dag = DAG(dag_id='test_dags', start_date=DEFAULT_DATE)
+        DummyOperator(task_id='dummy', dag=dag, owner='airflow')
+
+        session = settings.Session()
+
+        orm_dag = DagModel(
+            dag_id=dag.dag_id,
+            has_task_concurrency_limits=False,
+            next_dagrun=DEFAULT_DATE,
+            next_dagrun_create_after=DEFAULT_DATE + timedelta(days=1),
+            is_active=True,
+        )
+        assert orm_dag.max_active_runs is not None
+        session.add(orm_dag)
+        session.flush()
+
+        needed = DagModel.dags_needing_dagruns(session).all()
+        assert needed == [orm_dag]
+        # set to None and ensure it's not sent to scheduler
+        orm_dag.max_active_runs = None
+        session.merge(orm_dag)
+        session.flush()
+        needed = DagModel.dags_needing_dagruns(session).all()
+        assert needed == []
+
```

Comment:
```suggestion
```

I don't think we need this behaviour _and_ the explicit flag, so since we have added the flag we should just use that.

In which case we should also re-title this PR please.